app/controllers/spree/affirm_controller.rb

```diff
@@ -24,31 +24,7 @@ def confirm
         payment_method: payment_method
       )
 
-      # check if data needs to be updated
-      unless _affirm_checkout.valid?
-
-        _affirm_checkout.errors.each do |field, error|
-          case field
-          when :billing_address
-            # FIXME(brian): pass the phone number to the |order| in a better place
-            phone = order.bill_address.phone
-            order.bill_address = generate_spree_address(_affirm_checkout.details['billing'])
-            order.bill_address.phone = phone
-
-          when :shipping_address
-            # FIXME(brian): pass the phone number to the |order| in a better place
-            phone = order.shipping_address.phone
-            order.ship_address = generate_spree_address(_affirm_checkout.details['shipping'])
-            order.ship_address.phone = phone
-
-          when :billing_email
-            order.email = _affirm_checkout.details["billing"]["email"]
-
-          end
-        end
-
-        order.save
-      end
```

Comment:
Is there a reason we don't need all this anymore?

Reply:
Same as below, it wasn't being used. This was checking the address before sending it through so it didn't quite make sense